src/helpers/ResolveRundownIntoPlaylist.ts

```diff
@@ -59,3 +77,75 @@ export function ResolveRundownIntoPlaylist(
 
 	return { resolvedPlaylist, untimedSegments }
 }
+
+function shouldLookForGraphicProfile(segment: UnrankedSegment, rundown: ResolvedPlaylistRundown): boolean {
+	const isKlarOnAirSegment = /^klar[\s-]*on[\s-]air/i.test(segment.name ?? '')
+	const isFloated = segment.iNewsStory.meta.float ?? false
+	const hasGraphicProfile = rundown?.payload?.graphicProfile !== undefined
+	return !isFloated && isKlarOnAirSegment && !hasGraphicProfile
+}
+
+type UnsortedGraphicProfile = { cueIndex: number; graphicProfile: string }
+
+/**
+ *
+ * @param segment Segment of which graphic profile cues are to be extracted.
+ * @returns A sorted list of graphic profiles for the given segment.
+ */
+function getOrderedGraphicProfiles(segment: UnrankedSegment): string[] {
+	const graphicProfiles = getGraphicProfiles(segment)
+
+	const cueOrder = getCueOrder(segment)
+	const orderedGraphicProfiles = sortGraphicProfiles(graphicProfiles, cueOrder)
+
+	return orderedGraphicProfiles
+}
+/**
+ *
+ * @param graphicProfiles Graphic Profiles to sort.
+ * @param cueOrder The sort order
+ * @returns A sorted list of graphicProfiles
+ */
+function sortGraphicProfiles(graphicProfiles: UnsortedGraphicProfile[], cueOrder: number[]): string[] {
+	return cueOrder.reduce<string[]>((orderedGraphicsProfiles, cueId) => {
+		const graphicProfile = graphicProfiles.find(({ cueIndex }) => cueIndex === cueId)?.graphicProfile
+		return graphicProfile ? [...orderedGraphicsProfiles, graphicProfile] : orderedGraphicsProfiles
+	}, [])
+}
+
+/**
+ *
+ * @param segment Segment to extract graphic profiles from.
+ * @returns The list of graphic profiles along with their cue index.
+ */
+function getGraphicProfiles(segment: UnrankedSegment): UnsortedGraphicProfile[] {
```

Comment:
It's confusing that we have a method called "getGraphicProfiles" and another method called "sortGraphicProfiles" that returns two completely different types.